misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -62,6 +70,10 @@ internal class SqsJobConsumer @Inject internal constructor(
     private val running = AtomicBoolean(true)
 
     override fun run() {
+      // Don't call handlers until all services are ready, otherwise handlers will crash because the
+      // services they might need (databases, etc.) won't be ready.
+      serviceManagerProvider.get().awaitHealthy()
```

Comment:
why this approach as opposed to the original proposal in https://github.com/cashapp/misk/pull/1052/files ?

the original proposal seemed fine and it was more explicit about declaring what you depend on.

Reply:
I think it’s a mistake ’cause it means we’re running Jobs before the service is officially booted up. That probably means every job needs to know every service it depends on, which is weird.

(We wouldn’t want HTTP actions to be received when the webserver is running but before the database is ready)